### DISCUSSION TITLE
[Feature]: Full Customization for Spectral Modes (All, not just custom)

### DISCUSSION THREAD

**Discussion by ThePackDoctor** [gpstar81/GPStar-proton-pack#414] [Q&A]
### What would you like to add in terms of software changes?

I don't use the Holiday, Halloween or classic Spectral Modes, so I was wondering if it's possible the colors could be changeable in the custom flashing code? I'm interested in creating more than one custom spectral mode, for a few custom 2009-esque modes of my own design. I dunno how much of a pain this would be, since the classic spectral (and maybe the others, I've literally never even tested the other spectrals) actually changes colors, not just animations. Is it possible? Just a hypothetical. I'm alright with not. Maybe the option (Only added in custom flashing? I don't own the WebUI attenuator system yet, it's on the list.) to have more than one custom spectral, to keep the original spectral modes intact. (albeit off in my case) Thank you!!

### Would this request involve any specific hardware?

Existing hardware only; All RGBLEDs in the pack (and maybe now the wand with the new vent board)

### Homework Completed

- [X] I did my part!

**Comment by nomakewan** [comment]
If you mean sacrificing the non-custom modes to give yourself three additional custom modes, that is of course possible (in that it could be easily coded) but it is not something we will ever officially implement.

You are free to do this yourself on your own custom build, however. The animations are identical between all of these modes (they all act identical to Proton except for the color data), so you don't need to worry about that.

Look into the Neutrona Wand's `Colours.h` file and check colors like `C_REDGREEN` for the dual color modes or `C_RAINBOW` for the rainbow modes. That should give you an idea of how to shift colours.

But if you wanted to actually give yourself the ability to customize the colours of those other modes on the fly the same way you can with Spectral Custom, that would require significantly more work on your part.

I'm going to close this as not planned and move it to a discussion in case you would like to talk further about how you might go about doing this yourself.

**Reply by ThePackDoctor** [reply]
Thank you!! I did mean a more permanent option, so the help in which the colors shift is very helpful! Thank you! I can't remember, does the spectral mode set have a specific sound profile/font?

**Reply by nomakewan** [reply]
No, the Spectral and Spectral Custom modes use the same one as Proton. Only Holiday has unique sounds (as of the upcoming 5.4.3, that is).

**Comment by gpstar81** [comment]
I always imagined one day, one could add a unlimited (or limited) amount of extra firing modes over the web ui interface and customise the colours for them and the firing animations (preset ones that exist already). Also at the same time, one could disable or enable over the web interface which modes they want enabled or disabled (even the original firing modes, slime etc).
